### PR TITLE
[UI > Settings] Implement better searching

### PR DESCRIPTION
**Pull request by nexpid**
Allows searching by plugin/theme name, ignores whitespace characters

### CODE REVIEW

**Comment by Cynosphere** [comment]
Can you add `toLowerCase` searching to Asset Browser as well? Two birds with one generalized PR title.

**Comment by nexpid** [comment]
good idea

**Comment by nexpid** [comment]
@Cynosphere done <:thumbsupcat:1105493146529431633>

**Review by redstonekasi — APPROVED** [review]

File: src/ui/settings/pages/AssetBrowser.tsx [hunk 1]
```diff
@@ -11,18 +11,14 @@ export default function AssetBrowser() {
...
             <RN.View style={{ flex: 1 }}>
-                <Search
-                    style={{ margin: 10 }}
-                    onChangeText={(v: string) => setSearch(v)}
-                    placeholder="Search"
-                />
+                <Search style={{ margin: 10 }} onChangeText={(v: string) => setSearch(v[0] === "*" ? v : v.toLowerCase())} placeholder="Search" />
                 <RN.FlatList
-                    data={Object.values(all).filter(a => a.name.includes(search) || a.id.toString() === search)}
+                    data={Object.values(all).filter((a) => (search[0] === "*" ? a.name.includes(search.slice(1)) : a.name.toLowerCase().includes(search)) || a.id.toString() === search)}
```
Comment (redstonekasi) [inline src/ui/settings/pages/AssetBrowser.tsx:16]: In what case would case-sensitive searching be needed?

Comment (nexpid) [reply src/ui/settings/pages/AssetBrowser.tsx:16]: for the new tabs v2 icons

Comment (redstonekasi) [reply src/ui/settings/pages/AssetBrowser.tsx:16]: Isn't it perfectly fine to just also search them case-insensitive?

**Pull request closed without merging**